src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -176,7 +176,7 @@ PeerImp::run()
     if (auto const iter = headers_.find("Closed-Ledger");
         iter != headers_.end())
     {
-        closed = parseLedgerHash(iter->value());
+        closed = parseLedgerHash(std::string{iter->value()});
```

Comment:
I feel `const std::string&` provides a stronger guarantee of existence (no dangling pointer) of the data, rather than `std::string_view`. What would happen if the input data goes out of scope? I know that `iter->value()` does not face that issue, but I'm concerned about other calls to `base64_decode`.

Reference: Notes section of https://en.cppreference.com/w/cpp/string/basic_string_view

Reply:
Any existing caller of `base64_decode` is already passing a `std::string`. Changing `base64_decode` to accept a `std::string_view` is safe because there is no use of the parameter, or any alias to it, after the function returns.

Reply:
fixed in https://github.com/XRPLF/rippled/pull/4509/commits/2cf0eb896c64d078133f108fc1fe55d79c349391